backend/pluto/tests/_internal/server/test_flask_server_integr.py

```diff
@@ -1,54 +1,91 @@
+class TestFlaskServer:
+    @pytest.fixture
+    def server(self):
+        srv = make_flask_server(ConfigMock.parse(), DatabaseMock(ConfigMock.parse()))
+        return ParallelServer(srv)
 
-        self.server.run()
+    @pytest.fixture
+    def client(self):
+        srv = make_flask_server(ConfigMock.parse(), DatabaseMock(ConfigMock.parse()))
+        with srv.app.test_client() as c:
+            yield c
 
-    def down(self):
-        #Eu só achei um jeito de desligar o servidor: apertando ctr+c
+    def test_init(self, server):
         pass
 
-    # testing reqs
-
-    def action(self, name):
-        """
-        This function takes `name` argument and returns `Hello <name>`.
-        """
-        return "Hello " + name
-    
-    def request_add_random_expense(self):
-        dictToSend = {
-            "user_id": random.randint(0, 1000),
-            "src": "mercadao",
-            "amount": random.random() * 100,
-        }
-        res = requests.post("http://localhost:5000/expenses/", json=dictToSend)
-        return res.content
-
-    def request_add_random_income(self):
-        dictToSend = {
-            "user_id": random.randint(0, 1000),
-            "src": "salario",
-            "amount": random.random() * 10000,
-        }
-        print("chegou aqui ")
-        res = requests.post("http://localhost:5000/incomes/", json=dictToSend)
-        return res.content
-
-    #real tests
\ No newline at end of file
+    def test_run_stop(self, server):
+        server.run()
+        server.__delete__()
+
+    def test_run_stop(self, server):
+        server.run()
+        server.__delete__()
+
+    def test_upload_file_from_request(self, server):
+        class File:
+            def __init__(self, filename):
+                self.filename = filename
+
+            def save(self, filepath):
+                pass
+
+        class Request:
+            def __init__(self, filename):
+                self.files = dict(file=File(filename))
+
+        server._upload_file_from_request(Request("tests/fixtures/valid_incomes.csv"))
+
+    def test_list_incomes(self, client):
+        client.post("/incomes/", data={"user_id": "putin"})
+        incomes = client.get("/incomes/")
+        print(incomes)
+
+    def test_income_upload(self, client):
+        # TODO: Properly test file upload. I found an example at
+        # https://stackoverflow.com/questions/35684436/testing-file-uploads-in-flask
+        client.post("/upload/incomes/", data={})
+
+    def test_expense_upload(self, client):
+        # TODO: properly test file upload
+        client.post("/upload/expenses/", data={})
+
+    def test_list_expenses(self, client):
+        client.post("/expenses/", data={"user_id": "putin"})
+        expenses = client.get("/expenses/")
+        print(expenses)
+
+    def test_get_user(self, client):
+        client.post("/users/", data={"email": "putin@massacre.death"})
+        client.get("/users/", data={"email": "putin@massacre.death"})
+
```

Comment:
@Pendulun These tests are not working very well... But I found the stack overflow link I mentioned in the comment above, it seems to provide proper instructions on how to test the file upload.